module/dice/basic-roll.mjs

```diff
@@ -153,23 +170,47 @@ export default class BasicRoll extends Roll {
      * @param {BasicRollMessageConfiguration} message  Configuration for the roll message.
      * @returns {boolean}                              Explicitly return `false` to prevent rolls.
      */
-    for ( const hookName of hookNames ) {
+    for ( const hookName of config.hookNames ) {
       const name = `dnd5e.post${hookName.capitalize()}RollConfiguration`;
       if ( Hooks.call(name, rolls, config, dialog, message) === false ) return [];
     }
 
-    if ( config.evaluate === false ) return rolls;
-    for ( const roll of rolls ) await roll.evaluate();
+    return rolls;
+  }
 
+  /* -------------------------------------------- */
+
+  /**
+   * Stage two of the standard rolling workflow, evaluating the rolls.
+   * @param {BasicRoll[]} rolls                           Rolls to evaluate.
+   * @param {BasicRollProcessConfiguration} [config={}]   Configuration for the rolls.
+   * @param {BasicRollMessageConfiguration} [message={}]  Configuration for message creation.
+   */
+  static async buildEvaluate(rolls, config={}, message={}) {
+    if ( (config.evaluate !== false) ) {
+      for ( const roll of rolls ) await roll.evaluate();
+    }
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Stage three of the standard rolling workflow, posting a message to chat.
+   * @param {BasicRoll[]} rolls                      Rolls to evaluate.
+   * @param {BasicRollProcessConfiguration} config   Configuration for the rolls.
+   * @param {BasicRollMessageConfiguration} message  Configuration for message creation.
+   * @returns {ChatMessage5e|void}
+   */
+  static async buildPost(rolls, config, message) {
     message.data = foundry.utils.expandObject(message.data ?? {});
     const messageId = config.event?.target.closest("[data-message-id]")?.dataset.messageId;
     if ( messageId ) foundry.utils.setProperty(message.data, "flags.dnd5e.originatingMessage", messageId);
 
-    if ( rolls?.length && (message.create !== false) ) {
-      await this.toMessage(rolls, message.data, { rollMode: message.rollMode });
+    if ( rolls?.length && (config.evaluate !== false) && (message.create !== false) ) {
+      message.created = await this.toMessage(rolls, message.data, { rollMode: message.rollMode });
```

Comment:
Elsewhere and in the docs this is `document`. Not sure if this should be `message.document =` then, or if the other places should change to use `created` instead.